docs/extending/datagrabber.rst

```diff
@@ -272,152 +272,172 @@ And we can create our datagrabber:
+``BOLD_confounds`` element does not only consists of a ``path``, but it requries
+more information about the format of the confounds file. Thus, the
+``BOLD_confounds`` element is a dictionary with the following keys:
 
 - ``path``: the path to the confounds file.
-- ``format``: the format of the confounds file. Currently, this can be either ``fmriprep`` or ``adhoc``.
+- ``format``: the format of the confounds file. Currently, this can be either
+  ``fmriprep`` or ``adhoc``.
 
-The ``fmriprep`` format corresponds to the format of the confounds files generated by `fMRIPrep`_. The
-``adhoc`` format corresponds to a format that is not standardized.
+The ``fmriprep`` format corresponds to the format of the confounds files
+generated by `fMRIPrep`_. The ``adhoc`` format corresponds to a format that is
+not standardised.
 
 .. note::
-   The ``mappings`` key is only required if the ``format`` is ``adhoc``. If the ``format`` is ``fmriprep``, the
-   ``mappings`` key is not required.
 
+   The ``mappings`` key is only required if the ``format`` is ``adhoc``. If the
+   ``format`` is ``fmriprep``, the ``mappings`` key is not required.
 
-Currently, Junifer provides only one confound remover step
-(:class:`.fMRIPrepConfoundRemover`), which relies entirely on the ``fmriprep`` confound
-variable names. Thus, if the confounds are not in ``fmriprep`` format, the user will need to provide the mappings
-between the *ad-hoc* variable names and the ``fmriprep`` variable names.
-This is done by specifying the ``adhoc`` format and providing the mappings as a dictionary in the ``mappings`` key.
+Currently, junifer provides only one confound remover step
+(:class:`.fMRIPrepConfoundRemover`), which relies entirely on the ``fmriprep``
+confound variable names. Thus, if the confounds are not in ``fmriprep`` format,
+the user will need to provide the mappings between the *ad-hoc* variable names
+and the ``fmriprep`` variable names. This is done by specifying the ``adhoc``
+format and providing the mappings as a dictionary in the ``mappings`` key.
 
-In the following example, the confounds file has 3 variables that are not in the ``fmriprep`` format. Thus, we will
-provide the mappings for these variables to the ``fmriprep`` format.
+In the following example, the confounds file has 3 variables that are not in the
+``fmriprep`` format. Thus, we will provide the mappings for these variables to
+the ``fmriprep`` format. For example, the ``get_item`` method could look like
+this:
 
 .. code-block:: python
 
-   out["BOLD_confounds"]: {
-      "path": f"{subject}/{session}/func/{subject}_{session}_confounds.tsv",
-      "format": "adhoc",
-      "mappings": {
-         "fmriprep": {
-            "variable1": "rot_x",
-            "variable2": "rot_z",
-            "variable3": "rot_y",
-         }
-      },
-   }
-
+   def get_item(
+      self, subject: str, session: str
+   ) -> dict[str, str | dict[str, str | dict[str, dict[str, str]]]]:
```

Comment:
I removed the typing from the examples on purpose.

While it's better for the actual code, it just makes the example more complicated to read with this kind of things.

Reply:
I'll simplify it.